include/iceflow/producer.hpp

```diff
@@ -44,41 +45,34 @@ namespace iceflow {
  */
 class IceflowProducer {
 public:
-  IceflowProducer(std::shared_ptr<IceFlow> iceflow, const std::string &pubTopic,
-                  uint32_t numberOfPartitions,
-                  std::chrono::milliseconds publishInterval);
-
-  IceflowProducer(std::shared_ptr<IceFlow> iceflow, const std::string &pubTopic,
+  IceflowProducer(std::shared_ptr<ndn::svs::SVSPubSub> svsPubSub,
+                  const std::string &nodePrefix, const std::string &syncPrefix,
+                  const std::string &upstreamEdgeName,
+                  uint32_t numberOfPartitions);
+
+  IceflowProducer(std::shared_ptr<ndn::svs::SVSPubSub> svsPubSub,
+                  const std::string &nodePrefix, const std::string &syncPrefix,
+                  const std::string &upstreamEdgeName,
                   uint32_t numberOfPartitions,
-                  std::chrono::milliseconds publishInterval,
                   std::shared_ptr<CongestionReporter> congestionReporter);
 
   ~IceflowProducer();
 
   void pushData(const std::vector<uint8_t> &data);
 
-  void setPublishInterval(std::chrono::milliseconds publishInterval);
-
   void setTopicPartitions(uint64_t numberOfPartitions);
 
   uint32_t getProductionStats();
 
 private:
   IceflowProducer(
-      std::shared_ptr<IceFlow> iceflow, const std::string &pubTopic,
-      uint32_t numberOfPartitions, std::chrono::milliseconds publishInterval,
+      std::shared_ptr<ndn::svs::SVSPubSub> svsPubSub,
+      const std::string &nodePrefix, const std::string &syncPrefix,
+      const std::string &upstreamEdgeName, uint32_t numberOfPartitions,
       std::optional<std::shared_ptr<CongestionReporter>> congestionReporter);
 
   uint32_t getNextPartitionNumber();
 
-  QueueEntry popQueueValue();
-
-  bool hasQueueValue();
-
-  void resetLastPublishTimePoint();
-
-  std::chrono::time_point<std::chrono::steady_clock> getNextPublishTimePoint();
-
   void reportCongestion(CongestionReason congestionReason,
                         const std::string &edgeName);
```

Comment:
I am wondering how to best utilize the `reportCongestion` method with the new API – I guess we should probably move it to the main `IceFlow` class and then allow for registering a `CongestionReporter` there, right?

Reply:
Should be done via the recent commits. In the producer and consumer classes, we need to decide from where to call the congestion reporter if we want to use that feature.